test.js

```diff
@@ -6,100 +6,146 @@ const crypto = require('crypto')
 const Fastify = require('fastify')
 const fastifyKafka = require('./')
 
-const options = {
+const logger = { level: 'trace' }
+
+const defaultOptions = {
   producer: {
     'metadata.broker.list': '127.0.0.1:9092',
-    'group.id': 'kafka1',
     'fetch.wait.max.ms': 10,
     'fetch.error.backoff.ms': 50,
     'dr_cb': true
   },
   consumer: {
     'metadata.broker.list': '127.0.0.1:9092',
-    'group.id': 'kafka1',
     'fetch.wait.max.ms': 10,
     'fetch.error.backoff.ms': 50
+  },
+  consumerTopicConf: {
+    'auto.offset.reset': 'beginning'
   }
 }
 
-test('Should add a kafka producer and consumer to the fastify instance', t => {
-  t.plan(2)
-
-  const fastify = Fastify()
-  const group = crypto.randomBytes(20).toString('hex')
-  options.producer['group.id'] = group
+test('communication', t => {
+  t.plan(5)
+  const options = copyPlainObject(defaultOptions)
+  const group = generateGroupId()
   options.consumer['group.id'] = group
+  options.producer['group.id'] = group
 
-  fastify
-    .register(fastifyKafka, options)
+  const topicName = generateTopicName()
+
+  const producerFastify = Fastify({ logger })
+  const consumerFastify = Fastify({ logger })
+
+  t.tearDown(() => producerFastify.close())
+  t.tearDown(() => consumerFastify.close())
+
+  consumerFastify
+    .register(fastifyKafka, {...options, producer: undefined})
     .after(err => {
       t.error(err)
 
-      fastify.kafka.producer.on('error', t.fail)
-      fastify.kafka.consumer.on('error', t.fail)
+      consumerFastify.kafka.consumer.on('error', t.fail)
+      consumerFastify.kafka.subscribe(topicName)
 
-      fastify.kafka.subscribe('test')
-
-      fastify.kafka.on('test', (msg, commit) => {
+      consumerFastify.kafka.on(topicName, (msg, commit) => {
         t.strictEqual(msg.value.toString(), 'hello world!')
         commit()
-        fastify.close()
+
+        t.ok(true)
       })
 
-      fastify.kafka.consume()
+      consumerFastify.kafka.consume()
+    })
+
+  producerFastify
+    .register(fastifyKafka, {...options, consumer: undefined})
+    .after(err => {
+      t.error(err)
+
+      producerFastify.kafka.producer.on('error', t.fail)
       setTimeout(() => {
-        fastify.kafka.push({
-          topic: 'test',
+        producerFastify.kafka.push({
+          topic: topicName,
           payload: 'hello world!',
           key: 'testKey'
         })
+
+        t.ok(true)
       }, 1000)
```

Comment:
while you are refactoring, can you remove this timeout?

Reply:
done